fiftyone/core/dataset.py

```diff
@@ -204,7 +206,7 @@ def __init__(self, name=None, persistent=False, _create=True):
         self._deleted = False
 
     def __len__(self):
-        return self._sample_collection.count_documents({})
+        return self.aggregate(foa.Count()).count
```

Comment:
How about leveraging the power of aggregations by computing `len(self)` and `get_tags()` via a single `aggregate()` call in the definitions of `Dataset.summary()` and `DatasetView.summary()`? 😄

Reply:
Of course, on it.